oml/datasets/audios.py

```diff
@@ -0,0 +1,392 @@
+            start_times: List of start time offsets in `seconds` for each audio.
+            spec_repr_func: Spectral representation extraction function used for visualization.
+        """
+        assert (start_times is None) or (
+            len(paths) == len(start_times)
+        ), "The length of 'start_times' must match the length of 'paths' if 'start_times' is provided."
+        assert sample_rate > 0, "The sample rate must be a positive integer."
+        assert max_num_seconds > 0, "The maximum number of seconds must be a positive float."
+        assert isinstance(is_mono, bool), "'is_mono' must be a boolean."
+
+        paths = [Path(p) if dataset_root is None else Path(dataset_root) / p for p in paths]
+        assert all(
+            path.suffix in AUDIO_EXTENSIONS for path in paths
+        ), f"Input audios should have one of '{AUDIO_EXTENSIONS}' extensions."
+
+        if extra_data is not None:
+            assert all(
+                len(record) == len(paths) for record in extra_data.values()
+            ), "All the extra records need to have the size equal to the dataset's size"
+            self.extra_data = extra_data
+        else:
+            self.extra_data = {}
+
+        self.input_tensors_key = input_tensors_key
+        self.index_key = index_key
+
+        self._paths = paths
+        self._sample_rate = sample_rate
+        self._num_frames = int(max_num_seconds * sample_rate)
+        self._is_mono = is_mono
+        self._frame_offsets = (
+            [int(st * sample_rate) for st in start_times] if start_times is not None else [0] * len(paths)
+        )
+        self._spectral_function = spec_repr_func or default_spec_repr_func
+
+    def _downmix_and_resample(self, audio: FloatTensor, sample_rate: int) -> FloatTensor:
+        """
+        (Optionally) downmix audio to mono and resample it to the dataset's sampling rate.
+
+        Args:
+            audio: Input audio tensor.
+            sample_rate: Original sampling rate of the audio.
+
+        Returns:
+            Processed audio tensor.
+        """
+        from torchaudio.transforms import Resample
+
+        if self._is_mono and audio.shape[0] != 1:
+            audio = audio.mean(dim=1, keepdim=True)
+        if sample_rate != self._sample_rate:
+            resampler = Resample(sample_rate, self._sample_rate)
+            audio = resampler(audio)
+        return audio
+
+    def _trim_or_pad(self, audio: FloatTensor, frame_offset: int) -> FloatTensor:
+        """
+        Trim and/or pad the audio to match the desired number of frames.
+
+        Args:
+            audio: Audio tensor.
+            frame_offset: Frame offset for trimming.
+
+        Returns:
+            Trimmed and/or padded audio tensor.
+        """
+        if audio.shape[1] > self._num_frames:
+            audio = audio[:, frame_offset : frame_offset + self._num_frames]
```

Comment:
If you s`tart_time` is bigger than a length of an audio, you get array of zeros here, right?
The problem if this condition is met for 2 or more audios. In such case you have a set of the same inputs (which are simply zeros) having some different labels, which will negatively affect the training process.

Thus, I suggest to raise a error instead and add a test on that

Reply:
done